rainloop/lib/Controller/PageController.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace OCA\RainLoop\Controller;
+
+use OCA\RainLoop\Util\RainLoopHelper;
+
+use OCP\App\IAppManager;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\ContentSecurityPolicy;
+use OCP\AppFramework\Http\TemplateResponse;
+use OCP\IConfig;
+use OCP\IRequest;
+
+class PageController extends Controller {
+	private $userId;
+	private $config;
+	private $appManager;
+	private $appPath;
+
+	public function __construct($AppName, IRequest $request, IAppManager $appManager, IConfig $config) {
+		parent::__construct($AppName, $request);
+		$this->appPath = $appManager->getAppPath('rainloop');
+		$this->config = $config;
+		$this->appManager = $appManager;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function index() {
+		\OC::$server->getNavigationManager()->setActiveEntry('rainloop');
+
+		\OCP\Util::addStyle('rainloop', 'style');
+
+		$sUrl = RainLoopHelper::normalizeUrl(RainLoopHelper::getAppUrl());
+
+		$params = [
+			'rainloop-iframe-url' => RainLoopHelper::normalizeUrl($sUrl).'?OwnCloudAuth'
+		];
+
+		$response = new TemplateResponse('rainloop', 'index', $params);
+
+		return $response;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function appGet() {
+		$this->app();
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function appPost() {
+		$this->app();
+	}
+
+	public function app() {
+
+		$csp = new ContentSecurityPolicy();
+		// fixes object-src: 'none' which blocks chrome from preview pdf
+		$csp->addAllowedObjectDomain("'self'");
```

Comment:
My goal here was to use NC's built-in CSP functionality to replace the old manual `$policy = <long multi-line policy>; header($policy)` thing. You old comments seemed to indicate that the main reason for the manual CSP creation was for fixing previews of PDFs in Chrome.  I think the above does that. One thing I'm currently unsure of is whether `return $response` below is actually even doing anything. I really don't know what happens in the `include app/index.php`. I followed the chain of execution a bit into the Rainloop code itself, but realized it was futile to try to fully understand every code path initiated by the `include`. Ideally, we could use the `TemplateResponse()` better instead of an include here. Any thoughts on a way to make this happen? I'm also not 100% sure that the CSP is being set the way I intend here.